app/models/amusement_park.rb

```diff
@@ -1,3 +1,8 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics
+
+  def unique_name
+    mechanics.distinct.pluck(:name)
+  end
```

Comment:
I cannot imagine that this method actually works. I believe you should be getting an error saying that the relationship between an amusement park and a mechanic does not exist. This is unless you added that relationship yourself on the `mechanics` and `amusement_parks` tables.

In any case, the AR is correct if you got the `mechanics` connected properly.